app/exporters/contests/contest_exporter.rb

```diff
@@ -0,0 +1,100 @@
+require 'base_exporter'
+require 'csv'
+
+module Contests
+  class ContestExporter < Exporters::BaseExporter
+    include RenderAnywhere
+
+    def to_csv(contest, include_unranked_contestants)
+      problems = contest.problem_set.problems
+
+      scoreboard = CSV.generate do |csv|
+        csv << ["Rank", "User ID", "Real Name", "Username", "School", "Year"] + problems.map{ |p| p.name } + ["Total Score", "Time"]
+
+        num_ranked = 0
+        rank = nil
+        previous_record = nil
+
+        contest.scoreboard.each do |record|
```

Comment:
i think if you do a .includes(user: :school) you'll skip an N+1 accessing users and schools down in line 30

Reply:
hmm that might be a rails 4.2 feature actually..

Reply:
`scoreboard` is a custom method, not an ActiveRecord relation. It already has `.includes(:user)` inside it, but that might not load the schools... Should investigate if it's making lots of queries for the schools and if `.includes(user: :school)` fixes that.

Reply:
ah, cool, i see the sitch now. ty for the learning :)